UserTableViewController.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  UserTableViewController.swift
+//  Random Users
+//
+//  Created by Craig Belinfante on 9/13/20.
+//  Copyright © 2020 Erica Sadun. All rights reserved.
+//
+
+import UIKit
+
+class UserTableViewController: UITableViewController {
+    
+    let networkController = NetworkController()
+    let imageFetchQueue = OperationQueue()
+    
+    var users: [User] = []
+    var cache = Cache<String,Data>()
+    var operations: [String: FetchPhotoOperation] = [:]
+    
+    
+    
+    @IBAction func addButton(_ sender: UIBarButtonItem) {
+    }
+    
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    //    tableView.reloadData()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        networkController.fetchUsers { (_) in
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+
+    // MARK: - Table view data source
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return networkController.users.count
+    }
+
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "UserCell", for: indexPath) as? UserTableViewCell else {fatalError("Cannot deque cell")}
+
+        // Configure the cell...
+        cell.user = networkController.users[indexPath.row]
+        loadImage(forCell: cell, forItemAt: indexPath)
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didEndDisplaying cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+            operations[networkController.users[indexPath.item].name]?.cancel()
+        }
+
+    func loadImage(forCell cell: UITableViewCell, forItemAt indexPath: IndexPath) {
+       
+        // 💩💩💩💩💩💩💩💩
+        
+        let user = networkController.users[indexPath.item]
+        
+        if let cachedData = cache.value(forKey: user.thumbnail) {
+            cell.imageView?.image = UIImage(data: cachedData)
+          //  cell.textLabel?.text = user.name
+        } else {
+            let fetchOp = FetchPhotoOperation(user: user, image: .thumbnail)
+            let cachedImage = BlockOperation {
+                if let data = fetchOp.imageData {
+                    self.cache.cache(value: data, forKey: user.thumbnail)
+                }
+            }
+            let imageUpdated = BlockOperation {
```

Comment:
Minor nitpick: `imageUpdated` might be better named `updateImage` or even better `updateImageOperation`. Prefer descriptive names with swift where you can. It makes reading your code later (or someone else reading it) easier.